### ISSUE TITLE
Controller#xml_convert doesn't properly serialize nodes with text children

### ISSUE THREAD

**Issue by jeffhos** [chrisroberts/dav4rack#12]
If I return an element like this (from, say, get_property on my resource):

```
builder = Nokogiri::XML::Builder.new do | xml |
  xml.href "/user"
end
return builder.doc.root
```

it will be serialized as "<D:href/>" without the text content

It looks like this is because xml_convert doesn't serialize the text part element.text? returns false (which it does on the href element, since it's not a text node, even though it does have a text child)

**Comment by chrisroberts** [comment]
The convert method is a hold over and was badly updated. I've pushed up an xml_covert_fix branch that contains a very simple fix for this issue. It works as expected for me and I'd be happy to get a confirmation from you that this resolves your issue.

**Comment by chrisroberts** [comment]
Merged and now available in v0.2.4
